### DISCUSSION TITLE
Is it really impossible to bind a Open Generic Interface to an Open Generic Concrete Type?

### DISCUSSION THREAD

**Discussion by zadi0648** [hadashiA/VContainer#630] [Ideas]
```
builder.Register(
typeof(IReactiveProperty<>),
typeof(ReactiveProperty<>),
Lifetime.Singleton);
```

does not work because there is no signature
`Register(this IContainerBuilder, Type, Type, Lifetime)`
for `IContainerBuilder`

**Comment by reromanlee** [comment]
You'd need to specify nested type regardless, since it's logically impossible to have just empty Generic<> when it needs to be defined clearly for compiler to know what we're dealing with
